ngi_pipeline/engines/qc_ngi/launchers.py

```diff
@@ -0,0 +1,144 @@
+    """The main entry point for the qc pipeline."""
+    LOG.info("Processing project/sample {}/{}".format(project, sample))
+
+    # Two paths diverged in a yellow wood
+    project_analysis_path = os.path.join(project.base_path,
+                                         "ANALYSIS",
+                                         project.project_id,
+                                         "qc_ngi")
+    # and sorry I could not travel both
+    sample_analysis_path = os.path.join(project_analysis_path, sample.name)
+    # and be one traveler, long I stood
+    log_dir_path = os.path.join(project_analysis_path, "logs")
+    # and looked down one as far as I could
+    safe_makedir(sample_analysis_path)
+    # To where it bent in the undergrowth
+    safe_makedir(log_dir_path)
+    # I need to go to sleep
+
+    fastq_files_to_process = []
+    # I suppose I -should- have quoted the other one
+    src_fastq_base = os.path.join(project.base_path, "DATA",
+                                  project.project_id, sample.name)
+    # Whose woods these are I think I know
+    for libprep in sample:
+        # His house is in the village though
+        for seqrun in libprep:
+            # He will not see mt stopping here
+            for fastq_file in seqrun:
+                # To watch
+                path_to_src_fastq = os.path.join(src_fastq_base,
+                                                 libprep.name,
+                                                 seqrun.name,
+                                                 fastq_file)
+                # his woods
+                fastq_files_to_process.append(path_to_src_fastqc)
+    # fill up
+    paired_fastq_files = find_fastq_read_pairs(fastq_files_to_process).values()
+    # with snow
+    qc_cl_list = return_cls_for_workflow("qc", paired_fastq_files, sample_analysis_path)
+
+    sbatch_file_path = create_sbatch_file(qc_cl_list, project, sample, config)
+    try:
+        slurm_job_id = queue_sbatch_file(sbatch_file_path) 
+    except RuntimeError as e:
+        LOG.error('Failed to queue qc sbatch file for project/sample '
+                  '"{}"/"{}"!'.format(project, sample))
+    else:
+        LOG.info('Queued qc sbatch file for project/sample '
+                 '"{}"/"{}": slurm job id {}'.format(project, sample, slurm_job_id))
+
+
+
+def queue_sbatch_file(sbatch_file_path):
+    LOG.info("Queueing sbatch file {} for job {}".format(sbatch_outfile, job_identifier))
+    p_handle = execute_command_line("sbatch {}".format(sbatch_file_path),
+                                    stdout=subprocess.PIPE,
+                                    stderr=subprocess.PIPE)
+    p_out, p_err = p_handle.communicate()
+    try:
+        slurm_job_id = re.match(r'Submitted batch job (\d+)', p_out).groups()[0]
+    except AttributeError:
+        raise RuntimeError('Could not submit sbatch job for workflow "{}": '
+                           '{}'.format(job_identifier, p_err))
+    return int(slurm_job_id)
+
+
```

Comment:
I assume that this code is duplicated from the piper ngi, is it worth the effort to abstract the queue system with a command  like `submit_job` and implement couple of functions slurm related so if one day we switch to an SGE cluster we can adapt the code?